scripts/src/public/services/offers.ts

```diff
@@ -90,41 +86,42 @@ async function filterOffers(userId: string, app: Application | undefined, logger
 		language = acceptsLanguagesFunc(availableLanguages); // get the most suitable language for the client
 	}
 	return (await Promise.all(
-		app.offers
-			.map(async offer => {
-					if ((offerCounts.get(offer.id) || 0) >= offer.cap.per_user) {
-						return null;
-					}
-					const content = contents.get(offer.id);
-					if (!content) {
-						return null;
-					}
-					return offerDbToApi(offer, content, getOfferTranslations(language, offer.id, availableTranslations));
+		appOffers
+			.map(async appOffer => {
+				const offer = appOffer.offer;
+				if ((totalOfferCounts.get(offer.id) || 0) >= appOffer.cap.total) {
```

Comment:
We should add a default value to `Map#get`, like in python.

Reply:
ok. How do you suggest?

Reply:
Subclass or Proxy